man/loader.conf.xml

```diff
@@ -60,6 +60,12 @@
         selected entry will be stored as an EFI variable, overriding this option.
         </para>
 
+        <para>If set to <literal>auto-last-booted</literal> the selected boot entry will be saved
+        to <filename><replaceable>ESP</replaceable>/loader/last-booted</filename> and
+        automatically selected the next time systemd-boot starts. For this to work the file
+        must exist and be large enough. This can be done by using the following command:
+        <command>dd if=/dev/zero of=ESP/loader/last-booted bs=256 count=1</command></para>
```

Comment:
i really like the semantics of this, and that this requires both special preparation and a config option, that's just ugly.

what about doing it like this: 

1. no config setting.
2. but if `$ESP/loader/last-booted-xyz` exists then `xyz` is the last booted option
3. when booting an item abc and `$ESP/loader/last-booted-abc` already exists we won't do anything, otherwise we rename any file `last-booted-*` we find to this name.
4. We never create the file from the boot loader.

Thus, if someone wants this behaviour they just have to touch `$ESP/loader/last-botted`, and if they don't want it they have to rm `$ESP/loader/last-booted*`

You get the idea. 

This would mean, just like for the boot assessment stuff we just rely on renames, nothing else, a single "atomic" op basically, and people can opt into this with a simple unix tool: touch.

That all said, given this is something controlled by the user i probably would not do anything with the file system here, in this ase it should be fine to just update the efi var we have anyway. Updating efi vars on every boot sucks, but it's totally fine if we do it whenver the user changes their mind. And given that we nicely can suppress efi var updates when the user boots always the same menu item (and he apparently intends to, given he apparently turns on this option), we'd only induce efi var writes whenever the user *changes* their mind, and that should be totally ok hence.

hence, if you want this beavour, i'd propose to scrap this whole disk write stuff, and instead add a new special value to the existing "default" stanza in the config file: if set to `@last` or some other special string like that, then go only by efi var, and update the efi var on boot unless it already matches the same value. If the efi var isn't set, pick the newest, as usual.

Reply:
> hence, if you want this beavour, i'd propose to scrap this whole disk write stuff, and instead add a new special value to the existing "default" stanza in the config file: if set to @last or some other special string like that, then go only by efi var, and update the efi var on boot unless it already matches the same value. If the efi var isn't set, pick the newest, as usual.

Make sense.

Reply:
Changing the efi far sounds like a disaster to happen. Mind you that sometimes you have days where you flip between windows and linux several times a day. And even for a explicit opt-in that sounds dangerous. If you'd be fine, we could just trust the firmware to create (and grow) that file for us (It obviously would still be a opt-in). I certainly trust them more to get a bit of fat handling right than fucking up their nvram from too many write.